internal/source/source.go

```diff
@@ -169,13 +175,27 @@ func (s *Source) Open(ctx context.Context, position sdk.Position) error {
 		return fmt.Errorf("init jetstream iterator: %w", err)
 	}
 
+	// Async handlers & callbacks
+	conn.SetErrorHandler(internal.ErrorHandlerCallback(ctx))
+	conn.SetDisconnectErrHandler(internal.DisconnectErrCallback(ctx, func(c *nats.Conn) {
```

Comment:
I'm thinking about disconnects and reconnects in this way: disconnect = stop source iterator, reconnect = start source iterator. If that's correct, then I think that should be reflected through code, so we call Iterator.Stop() here.

Reply:
Now that [we move from storing the consumer sequence to storing the stream sequence](https://github.com/conduitio-labs/conduit-connector-nats-jetstream/pull/68/commits/7c53489a1b3ccafb57e36c93ead80f1b791e636c), that is now correct. Send a PR to fix it.

Reply:
Gotcha... I believe that those duplicates would be a surprise and an unexpected behavior. Connectors generally do their best to just resume from where they stopped. Do we have to work with the stream sequence in order to fix the bug?

Reply:
The acceptance tests bug to be more precise. Using the consumer sequence can add duplication (because Jetstream's expectation is at least once, and it will send the extras message even if messages are in flight in case of network failures, and It will also reset the consumer sequence in case of reconnection (they keeping the already ack stream sequence number in the correct place)

Reply:
And, duplicated records can still exist because Jetstream consumers will only guarantee at least once delivery.